arangodb-net-standard/DocumentApi/DocumentApiClient.cs

```diff
@@ -286,5 +287,54 @@ public async Task<DeleteDocumentsResponse<T>> DeleteDocumentsAsync<T>(string col
             }
         }
 
+        /// <summary>
+        /// Partially updates the document identified by document-handle.
+        /// The body of the request must contain a JSON document with the
+        /// attributes to patch(the patch document). All attributes from the
+        /// patch document will be added to the existing document if they do not
+        /// yet exist, and overwritten in the existing document if they do exist
+        /// there.
+        /// PATCH/_api/document/{document-handle}
+        /// </summary>
+        /// <param name="collectionName"></param>
+        /// <param name="documentKey"></param>
+        /// <param name="body"></param>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public async Task<PatchDocumentResponse> PatchDocumentAsync(string collectionName, string documentKey, object body, PatchDocumentQuery query = null)
+        {
+            return await PatchDocumentAsync(WebUtility.UrlEncode(collectionName) + "/" + WebUtility.UrlEncode(documentKey), body, query);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="documentId"></param>
+        /// <param name="body"></param>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public async Task<PatchDocumentResponse> PatchDocumentAsync(string documentId, object body, PatchDocumentQuery query = null)
```

Comment:
I think we should implement a template parameter to define the object type and make this: `PatchDocumentAsync<T>(string documentId, T body, PatchDocumentQuery query = null)`

We could then also include an overload method that uses object for convenience:

```
public async Task<PatchDocumentResponse> PatchDocumentAsync(string documentId, object body, PatchDocumentQuery query = null) {
    return PatchDocumentAsync<object>(documentId, body, query);
}
```

Reply:
There's some complication here since we might not want the type of a returned object (under `New` or `Old`) to be the same type as the update model. Should we support two type parameters then? @DiscoPYF any thoughts?